### PR TITLE
Backwards compatibility fix on tx interceptor for gaslimit value

### PR DESCRIPTION
**Pull request by iulianpascalau**
- removed backwards compatibility issue on transactions at interceptor level
- added protection on the API for transactions with higher gas limit than the transaction processor can cope

### CODE REVIEW

**Review by SebastianMarian — APPROVED** [review]

File: process/economics/apiEconomicsData.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	if data == nil {
+		return nil, process.ErrNilEconomicsData
+	}
+
+	return &apiEconomicsData{
+		economicsData: data,
+	}, nil
+}
+
+// CheckValidityTxValues checks if the provided transaction is economically correct. It overloads the original method
```
Comment (bogdan-rosianu) [inline process/economics/apiEconomicsData.go:23]: `if the provided transaction is economically correct` => `if the provided transaction respects the gas limit restrictions`

File: process/economics/apiEconomicsData_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/economics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewAPIEconomicsData(t *testing.T) {
+	t.Parallel()
```
Comment (bogdan-rosianu) [inline process/economics/apiEconomicsData_test.go:17]: `t.Parallel` on each inner test, as right now the inner tests will be run serial

File: process/economics/apiEconomicsData_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	minGasLimit := uint64(12)
+	maxGasLimitPerBlock := uint64(42)
+	args.Economics.FeeSettings.GasLimitSettings[0].MaxGasLimitPerMiniBlock = fmt.Sprintf("%d", maxGasLimitPerBlock)
+	args.Economics.FeeSettings.GasLimitSettings[0].MaxGasLimitPerMetaMiniBlock = fmt.Sprintf("%d", maxGasLimitPerBlock*10)
+	args.Economics.FeeSettings.MinGasPrice = fmt.Sprintf("%d", minGasPrice)
+	args.Economics.FeeSettings.GasLimitSettings[0].MinGasLimit = fmt.Sprintf("%d", minGasLimit)
+	economicsData, _ := economics.NewEconomicsData(args)
+	apiEconomicsData, _ := economics.NewAPIEconomicsData(economicsData)
+
+	t.Run("maximum gas limit as defined should work", func(t *testing.T) {
```
Comment (bogdan-rosianu) [inline process/economics/apiEconomicsData_test.go:49]: `t.Parallel()` on each inner test

File: process/economics/apiEconomicsData_test.go (hunk 2 above)
Comment (iulianpascalau) [reply process/economics/apiEconomicsData_test.go:17]: please check. Can not be done.

File: process/economics/apiEconomicsData_test.go (hunk 3 above)
Comment (iulianpascalau) [reply process/economics/apiEconomicsData_test.go:49]: please check. Can not be done.

File: process/economics/apiEconomicsData.go (hunk 1 above)
Comment (iulianpascalau) [reply process/economics/apiEconomicsData.go:23]: actually it also checks other stuff other than the gas limit as it calls the overloaded method

**Review by bogdan-rosianu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.
Errors are safe and will be solved soon. 
image

**Pull request merged**